usdigitalresponse/usdr-gost
# Define Terraform infrastructure for email metrics

### Ticket #2879 
## Description
See ticket for details. I followed the implementation plan very closely. I think the only difference is that I'm not passing `SES_CONFIGURATION_SET_DEFAULT` into the `consume_grants` container because that task is not otherwise configured to send email.

Note that this branch is based on the branch in https://github.com/usdigitalresponse/usdr-gost/pull/2945, which I was hoping to get merged first. That PR isn't required for these changes to work and I'm willing to rebase on `main` but I would like that PR to be reviewed. 🙂 

## Screenshots / Demo Video
N/A

## Testing
Once this is deployed to staging, it should be testable through the AWS console:
1. Go to the SNS page and subscribe to the `datadog-forwarder` topic. I did this with an email subscription.
2. Go to the SES page and click on the "staging.grants.usdr.dev" identity.
3. Click "Send Test Email".
4. On the "Send Test Email" page, select 'gost-staging-default' under "Configuration Set"
5. Switch from "Formatted" to "Raw".
6. Select "Custom" under "Scenario" and enter your email address under "Custom Recipient".
7. Enter this text in the "Message" box:
```
Subject: Amazon SES Raw Email Test
MIME-Version: 1.0
Content-Type: text/html

<!DOCTYPE html>
<html>
<body>
<p>Here is a formatted link: <a href="https://staging.grants.usdr.dev/grants/113293">Go to grant</a>.</p>
</body>
</html>
```
8. Fill out any other required fields and send the test email.

Now you can confirm:

- You received the email.
- The link in the email has a url that starts with `https://mail-redirect.us-west-2.staging.grants.usdr.dev`. Click on the link.
- You received several "Amazon SES Email Event Notification" notifications (through whatever subscription channel you set up) - a Send event, a Delivery event, an Open event and a Click event.

All of this worked for me after running `terraform apply` against my personal AWS account, setting the `website_domain_name` variable to a domain that I own and that I verified in SES.

### Automated and Unit Tests

### Manual tests for Reviewer

## Checklist
